client/components/data/query-domain-owner-username/index.ts

```diff
@@ -0,0 +1,51 @@
+import { useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import useUsersQuery from 'calypso/data/users/use-users-query';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { InfiniteData } from 'react-query';
+
+type User = {
+	login: string;
+	linked_user_ID: number;
+};
+
+type UsersData = {
+	users: User[];
+};
+
+export function useDomainOwnerUserName(
+	selectedSite: SiteDetails | null | undefined,
+	domain: ResponseDomain | null | undefined
+): string {
+	useQuerySitePurchases( selectedSite?.ID ?? -1 );
+
+	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
+
+	const selectedSubscriptionId = domain?.subscriptionId ?? '0';
+
+	const domainSubscription = purchases.filter(
+		( purchase ) => purchase.id === parseInt( selectedSubscriptionId )
+	)[ 0 ];
+
+	const { data, isLoading } = useUsersQuery(
+		selectedSite?.ID,
+		{},
+		{
+			enabled: domainSubscription !== undefined,
+		}
+	);
+
+	if ( isLoading || ! domainSubscription ) {
+		return '';
+	}
+
+	const teams = data as InfiniteData< UsersData > & UsersData;
+	const ownerUser = teams.users?.filter(
+		( user ) => user.linked_user_ID === domainSubscription?.userId
```

Comment:
```suggestion
		( user ) => user.ID === domainSubscription?.userId
```
In my testing, the user's ID property was `ID` rather than `linked_user_ID`. Could the property names vary, or should this be `user.ID`?

Reply:
@AllTerrainDeveloper and I spent some time troubleshooting this and we found that the endpoint returns:
* `ID` for simple sites
* both `linked_user_ID` (`id_wpcom` for v2 endpoint) and `ID` for atomic sites, where `linked_user_ID` is the actual user ID and `ID` is the Jetpack external user id

Dani [updated the code](https://github.com/Automattic/wp-calypso/pull/68158/files#diff-2ea6652302a0be15504ab269dde396fa7fc4353df0e5d8fb3f47324a32955129R47) to first check for `linked_user_ID`, if it it's undefined, then fallback to `ID`.

(See related thread here: p1664277912392969/1664274185.306699-slack-C01TJT42PMM)

Reply:
Just making a quick note that I also saw the same thing [here](https://github.com/Automattic/wp-calypso/blob/4f40f3ff0d18598384d7a6a9c924aa929c5cb6e3/client/my-sites/domains/domain-management/transfer/transfer-to-other-user/transfer-domain-to-other-user.jsx#L37).

Might be out of scope for this PR, but perhaps it would be useful to have a utility function to retrieve the user ID?